src/index.ts

```diff
@@ -21,44 +25,85 @@ async function run(): Promise<void> {
   setVariable(EnvironmentVariables.CI_BUILD_ID, buildId);
   setVariable(EnvironmentVariables.CI_BUILD_APPROVED, 'false');
 
-  setVariable(EnvironmentVariables.CI_COMMIT_ID, context?.sha);
-
-  const headCommit = event?.head_commit;
-  const author = headCommit?.author;
-  setVariable(EnvironmentVariables.CI_COMMIT_MESSAGE, headCommit?.message);
-  setVariable(EnvironmentVariables.CI_COMMITTER_USERNAME, author?.username);
-  setVariable(EnvironmentVariables.CI_COMMITTER_EMAIL, author?.email);
-  setVariable(EnvironmentVariables.CI_COMMITTER_NAME, author?.name);
+  const event = github.context.payload;
+  if (github.context.eventName === 'push') {
+    const pushPayload = event as PushEvent;
+    core.info(`The head commit is: ${pushPayload.head_commit}`);
+    populatePushEventCommitDetails(pushPayload, context);
+  } else if (github.context.eventName === 'pull_request') {
+    const pullRequestPayload = event as PullRequest;
```

Comment:
Should we note the effective HEAD commit here too?

Reply:
Meant to remove this comment while cleaning up.